src/Core/SemanticVersion.cs

```diff
@@ -39,32 +45,49 @@ public SemanticVersion(int major, int minor, int build, string specialVersion)
         {
         }
 
+        public SemanticVersion(int major, int minor, int build, string specialVersion, string metadata)
+            : this(new Version(major, minor, build), specialVersion, metadata)
+        {
+        }
+
         public SemanticVersion(Version version)
             : this(version, String.Empty)
         {
         }
 
         public SemanticVersion(Version version, string specialVersion)
-            : this(version, specialVersion, null)
+            : this(version, specialVersion, metadata: null, originalString: null)
         {
         }
 
-        private SemanticVersion(Version version, string specialVersion, string originalString)
+        public SemanticVersion(Version version, string specialVersion, string metadata)
+         : this(version, specialVersion, metadata, originalString: null)
+        {
+        }
+
+        private SemanticVersion(Version version, string specialVersion, string metadata, string originalString)
         {
             if (version == null)
             {
                 throw new ArgumentNullException("version");
             }
+
             Version = NormalizeVersionValue(version);
             SpecialVersion = specialVersion ?? String.Empty;
-            _originalString = String.IsNullOrEmpty(originalString) ? version.ToString() + (!String.IsNullOrEmpty(specialVersion) ? '-' + specialVersion : null) : originalString;
+            Metadata = metadata;
+
+            _originalString = String.IsNullOrEmpty(originalString) ? version.ToString() 

```

Comment:
The original string is used for getting the version component inside the class, it may not be the greatest design, but rewriting the class to same a string seems unnecessary for this legacy code.